res/app/device-list/details/device-list-details-directive.js

```diff
@@ -277,6 +279,20 @@ module.exports = function DeviceListDetailsDirective(
         }
       })()
 
+      // check if new column needs a new scope or not
+      // and build accordingly
+      function buildColumn(col, device) {
```

Comment:
There are people/companies with hundreds of connected devices, so I'd suggest testing with at least 500 fake devices, possibly even more. Ideally, though, we'd get rid of the scope entirely, or create and get rid of it on demand.